README.md

```diff
@@ -1,2 +1,319 @@
+
+TODO: what about title or icon? They should potentially also be accomodated, in a similar fashion.
+
+Thus, the following would all work well out of the box:
+
+```html
+<std-toast>Hello world!</std-toast>
+
+<std-toast><p>Hello world!</p></std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <button>Click me!</button>
+</std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <a href="https://example.com/" target="_blank">Click me!</button>
+</std-toast>
+```
+
+These can equivalently be created via JavaScript:
+
+```js
+// TODO: write this once we figure out the JS API for supplying HTML messages,
+// and for supplying customizable actions
+```
+
+More complex toasts,
+that don't fit the above content model,
+would require more custom handling on the part of the developer:
+
+```html
+<std-toast>
+  <p>Hello world!</p>
+  <p>A second paragraph?!</p>
+
+  <button>Action 1</button>
+  <button>Action 2</button>
+</std-toast>
+```
+
+Such an unusual toast would still integrate with other toasts in terms of stacking and positioning behavior,
+and some of the default styles that are inherited may be useful.
+But the page author will need to handle the action clicks themselves (instead of using the `"actionclick"` event),
+and will need to add additional styling to handle the extra contents.
+
+TODO: when we have a prototype, link to/show an example of this in action.
+
+#### Methods
+
+The `.show(options)` function shows a toast element,
+by toggling its `open=""` attribute to true.
+The `options` include:
+
+- `duration`: how long to show the toast, in milliseconds. Defaults to ???
+- `multiple`: TODO
+- `newestOnTop`: TODO
+
+TODO: how do `multiple` and `newestOnTop` work?
+Should those be a global setting?
+Per container?
+How do we deal with different toasts having different values?
+
+#### Events
+
+A `<std-toast>` element can fire the following events:
+
+- `"open"`: the toast was shown
+- `"close"`: the toast was closed, either explicitly by the user, or via the timeout.
+  (Note: if animations were applied, the toast may not be entirely invisible at the time this event fires)
+    - TODO: should we consider separate events for the start and end of any close animation?
+      This seems hard to do correctly if the user customizes the animation, though.
+- `"actionclick"`: the toast's call-to-action button or link was clicked, if one exists TODO
+
+### `showToast(message, options)`
+
+The `"std:elements/toast"` module also exports a convenience function,
+`showToast()`,
+which allows creating and showing toasts entirely from JavaScript.
+Behind the scenes,
+`showToast()` creates a `<std-toast>`,
+sets it up using the given message and options,
+inserts it as the last child of `<body>`,
+and then adds the `open=""` attribute,
+to make the toast visible.
+Finally,
+it returns the created `<std-toast>` element,
+allowing further manipulation by script.
+
+TODO: will `showToast()` support putting the toast into a specified container,
```

Comment:
How about returning the element so that script can immediately insert it somewhere else?

Reply:
This seems to be covered above, by

> Finally,
> it returns the created `<std-toast>` element,
> allowing further manipulation by script.

Reply:
OK, so do we need this TODO?

Reply:
We don't need the TODO, that behavior can be recreated by the developer, and we have a demo of how in the Common Patterns section. Removing the TODO now.